storey/drivers.py

```diff
@@ -501,3 +509,450 @@ async def _get_item(self, container, table_path, key, attributes):
+                 webapi: Optional[str] = None,
+                 aggregation_attribute_prefix: str = 'aggr_',
+                 aggregation_time_attribute_prefix: str = '_'):
+
+        NeedsRedisAccess.__init__(self, webapi)
+        self._redis = redis_client
+        self._key_prefix = key_prefix if key_prefix else self.DEFAULT_KEY_PREFIX
+        self._type = redis_type
+        self._mtime_name = '$_mtime_'
+
+        self._aggregation_attribute_prefix = aggregation_attribute_prefix
+        self._aggregation_time_attribute_prefix = aggregation_time_attribute_prefix
+        self._aggregation_prefixes = (self._aggregation_attribute_prefix,
+                                      self._aggregation_time_attribute_prefix)
+
+    @property
+    def redis(self):
+        if self._redis:
+            return self._redis
+        if self._type is RedisType.STANDALONE:
+            return redis.Redis.from_url(self._webapi_url, decode_responses=True)
+        self._redis = rediscluster.RedisCluster.from_url(self._webapi_url, decode_response=True)
+        return self._redis
+
+    def make_key(self, *parts):
+        return f"{self._key_prefix}{''.join([str(p) for p in parts])}"
+
+    @staticmethod
+    def _static_data_key(redis_key_prefix):
+        """
+        The Redis key for the Hash containing static data (AKA, "additional data").
+
+        The `redis_key_prefix` string should be the Redis key prefix used to store
+        data for a Storey container, table path, and key combination.
+        """
+        return f"{redis_key_prefix}:static"
+
+    def _aggregation_time_key(self, redis_key_prefix, feature_name):
+        """
+        The Redis key containing the associated timestamp for an aggregation.
+
+        The `redis_key_prefix` string should be the Redis key prefix used to store
+        data for a Storey container, table path, and key combination.
+        """
+        return f"{redis_key_prefix}:{self._aggregation_time_attribute_prefix}{feature_name}"
+
+    @staticmethod
+    def _list_key(redis_key_prefix, list_attribute_name):
+        """
+        The key containing a list of aggregation data.
+
+        The `redis_key_prefix` string should be the Redis key prefix used to store
+        data for a Storey container, table path, and key combination.
+        """
+        return f"{redis_key_prefix}:{list_attribute_name}"
+
+    @classmethod
+    def _convert_python_obj_to_redis_value(cls, value):
+        if isinstance(value, datetime):
+            if pd.isnull(value):
+                return None
+            return f'{cls.DATETIME_FIELD_PREFIX}{value.timestamp()}'
+        elif isinstance(value, timedelta):
+            return f'{cls.TIMEDELTA_FIELD_PREFIX}{value.total_seconds()}'
+        else:
+            if isinstance(value, float):
+                if value == math.inf or value == -math.inf:
+                    # Use the shorter infinity form (inf, -inf) to save space.
+                    value = f'{str(value)}'
```

Comment:
`str` and `f'...'` are redundant.
```suggestion
                    value = str(value)
```

Reply:
Agree

Reply:
fixed